src/python/WMCore/WMSpec/WMWorkload.py

```diff
@@ -362,15 +362,18 @@ def listTasksOfType(self, ttype):
         """
         return [t for t in self.taskIterator() if t.taskType() == ttype]
 
-    def getAllTasks(self):
+    def getAllTasks(self, cpuOnly=False):
         """
         _getAllTasks_
 
-        Get all tasks from a workload
+        Get all tasks from a workload.
+        If cpuOnly flag is set to True, then don't return utilitarian tasks.
         """
         tasks = []
-        pathNames = self.listAllTaskPathNames()
-        for n in pathNames:
+        for n in self.listAllTaskPathNames():
+            task = self.getTaskByPath(taskPath=n)
+            if cpuOnly and task.taskType() in ["Merge", "Harvesting", "Cleanup", "LogCollect"]:
+                continue
             tasks.append(self.getTaskByPath(taskPath=n))
```

Comment:
This can be ''' tasks.append(self.getTaskByPath(taskPath=n)) ''', right? So we don't have call the function again.